okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/OptionalMethod.java

```diff
@@ -169,6 +169,12 @@ private Method getMethod(Class<?> clazz) {
   private static Method getPublicMethod(Class<?> clazz, String methodName, Class[] parameterTypes) {
     Method method = null;
     try {
+      if (clazz == null) {
+        return null;
+      }
+      if ((clazz.getModifiers() & Modifier.PUBLIC) == 0) {
+        return getPublicMethod(clazz.getSuperclass(), methodName, parameterTypes);
+      }
```

Comment:
Off the top of my head I don't remember the the exact function that caused the error.

However, if remove the changes to ```OptionalMethod.java```, I get the following result.

```
io.grpc.okhttp.OptionalMethodTest > invokeOptional FAILED
    java.lang.AssertionError: expected:<testArg> but was:<null>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:743)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at io.grpc.okhttp.OptionalMethodTest.invokeOptional(OptionalMethodTest.java:96)

io.grpc.okhttp.OptionalMethodTest > isSupported FAILED
    java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at org.junit.Assert.assertFalse(Assert.java:74)
        at io.grpc.okhttp.OptionalMethodTest.isSupported(OptionalMethodTest.java:85)
```

The issue as I understand it only pops up when attempting to do reflection:
http://stackoverflow.com/questions/6161790/illegalaccessexception-when-using-call-dynamic-methods